File: livebox/src/main/java/com/creations/livebox/serializers/Serializer.java
package com.creations.livebox.serializers;

import java.lang.reflect.Type;

import okio.BufferedSource;

/**
 * @author Sérgio Serra on 25/08/2018.
 * Criations
 * sergioserra99@gmail.com
 */
public interface Serializer {
    <T> BufferedSource serialize(T input, Type type);

    <T> T deserialize(BufferedSource source, Type type);
}

File: livebox-common/src/main/java/com/creations/livebox_common/serializers/Serializer.kt
package com.creations.livebox_common.serializers

import okio.BufferedSource
import java.lang.reflect.Type

/**
 * @author Sérgio Serra
 * sergioserra99@gmail.com
 */
interface Serializer {
    fun <T> serialize(input: T, type: Type): BufferedSource?

    fun <T> deserialize(source: BufferedSource, type: Type): T?
}
